### ISSUE TITLE
Dune: "make -f Makefile.dune refman-html" complains about extra files, "make clean" doesn't fix

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#14194]
Can we get this fixed sooner rather than later?

```
$ make clean
$ make -f Makefile.dune refman-html
dune build @refman-html
Error: Multiple rules generated for
_build/default/user-contrib/Ltac2/.Bool.aux:
- file present in source tree
- <internal location>
Hint: rm -f user-contrib/Ltac2/.Bool.aux
Makefile.dune:101: recipe for target 'refman-html' failed

$ find . -name "*.aux" -exec rm {} \;

$ make -f Makefile.dune refman-html
dune build @refman-html
Error: Multiple rules generated for _build/default/doc/unreleased.rst:
- file present in source tree
- doc/dune:1
Hint: rm -f doc/unreleased.rst
Makefile.dune:101: recipe for target 'refman-html' failed
make: *** [refman-html] Error 1
```

**Comment by ejgallego** [comment]
You are mixing the two systems (make-based / dune-based) so trouble is expected; please post a reproduction from a pristine tree and that doesn't involve non-supported `make / dune` interleaved calls.

As stated in several places, I'd recommend that developers just set `COQ_USE_DUNE` which avoids the problem of unsupported interleaving.

[We could fix this using promote, but IMHO not worth it]

**Comment by SkySkimmer** [comment]
>As stated in several places, I'd recommend that developers just set COQ_USE_DUNE which avoids the problem of unsupported interleaving.

AFAICT people are deliberately interleaving, setting the variable just changes which calls they use -f with.

**Comment by Zimmi48** [comment]
IMHO the interleaving use case should be supported, at least for a reasonable number of targets (I understand this cannot be the case for the `test-suite` target).

That being said, this bug is not reproducible if you start from an empty state (obviously). I don't know from which state @jfehrle started but I expect that this is related to leftover outputs from before pulling in the new changes to the build system (and refusal to use `git clean`). If that's indeed the case, that's a very well known issue that has hit many contributors in the past and is also unfixable (though we shouldn't hit it anymore or not as frequently with the new hygienic builds).

Please reopen if you have a reproducible sequence of commands starting from a clean state.

**Comment by ejgallego** [comment]
We will support the test-suite soon, just not now; for docs, if we would update the make-based build so it generates the output in _build_vo that should do the trick mostly, current problem is indeed related to contamination of the source tree.

**Comment by jfehrle** [comment]
> As stated in several places, I'd recommend that developers just set COQ_USE_DUNE which avoids the problem of unsupported interleaving.

But it's not stated in `build-system.dune.md`, which should be the first place anyone should look for such information.  Who can track a blizzard of GitHub comments, emails, Zulip.  Just for your own sanity, I think you'd want to update that document rather than replying to endless questions one by one.

**Comment by jfehrle** [comment]
My complaint is that `make clean` no longer removes the un-"hygenic" files from the tree as it used to, e.g. "*.aux" files, which would make it fairly easy to recover from running the old make (whether intentionally or by mistake).  As it is, it's tedious.  And of course the problem is most visible as people transition to Dune.  The removal seems foolish.

**Comment by SkySkimmer** [comment]
Why do we even have make clean? git clean is better.

**Comment by Zimmi48** [comment]
> But it's not stated in `build-system.dune.md`, which should be the first place anyone should look for such information.

There's a PR open doing just that: #14184. Unfortunately it hasn't been finalized / merged yet.

>Why do we even have make clean? git clean is better.

`git clean` is terribly dangerous for non-expert users though because it removes untracked files. And note that `make clean` can also be used in a non-git copy.

> My complaint is that `make clean` no longer removes the un-"hygenic" files from the tree as it used to, e.g. "*.aux" files, which would make it fairly easy to recover from running the old make (whether intentionally or by mistake). As it is, it's tedious. And of course the problem is most visible as people transition to Dune. The removal seems foolish.

We tend to remove clean targets at the same time as we remove what generated these files. But maybe this was bad practice and should always be done in a separate PR to be merged later. Anyway, in this case I think what's done is done. We won't have this problem much in the future given that we are moving to more hygienic builds (even more so after #14212).
